source/extensions/filters/udp/udp_proxy/udp_proxy_filter.cc

```diff
@@ -351,6 +351,13 @@ void UdpProxyFilter::ActiveSession::write(const Buffer::Instance& buffer) {
   //       set. We allow the OS to select the right IP based on outbound routing rules if
   //       use_original_src_ip_ is not set, else use downstream peer IP as local IP.
   const Network::Address::Ip* local_ip = use_original_src_ip_ ? addresses_.peer_->ip() : nullptr;
+  if (!connected_) {
```

Comment:
seems you never set the `connected_` as true

Reply:
Oops, fixed.

Reply:
Would you be able to add test for it?

Reply:
Update unit test to ensure that `connect` will happen once and only once.

Reply:
great, thanks, my last request is just add comment here explain why use the `connect()` and why didn't use `connect()` when `use_original_src_ip_`